server/src/test/scala/io/delta/sharing/server/DeltaSharingServiceSuite.scala

```diff
@@ -1930,6 +1941,98 @@ class DeltaSharingServiceSuite extends FunSuite with BeforeAndAfterAll {
     assert(expectedMetadata == actions(1).metaData)
   }
 
+  object StreamingNotnullToNull {
+    val metadataV0 = Metadata(
+      id = "1e2201ff-12ad-4c3b-a539-4d34e9e36680",
+      format = Format(),
+      schemaString = """{"type":"struct","fields":[{"name":"name","type":"string","nullable":false,"metadata":{}}]}""",
+      configuration = Map("enableChangeDataFeed" -> "true"),
+      partitionColumns = Nil
+    )
+    val metadataV2 = metadataV0.copy(
+      schemaString = """{"type":"struct","fields":[{"name":"name","type":"string","nullable":true,"metadata":{}}]}"""
+    )
+  }
+
+  def getMetadataResponse(version: Long): String = {
+    readNDJson(
+      requestPath(s"/shares/share8/schemas/default/tables/streaming_notnull_to_null/metadata?version=$version"),
+      expectedTableVersion = Some(version)
+    )
+  }
+
+  def verifyMetadata(version: Long, expectedMetadata: Metadata): Unit = {
+    val response = getMetadataResponse(version)
+    val Array(protocol, metadata) = response.split("\n")
+    val expectedProtocol = Protocol(minReaderVersion = 1)
+    assert(expectedProtocol.wrap == JsonUtils.fromJson[SingleAction](protocol))
+    assert(expectedMetadata.wrap == JsonUtils.fromJson[SingleAction](metadata))
+  }
```

Comment:
Can we take `val response = getMetadataResponse(..)` out of `verifyMetadata()` and make `verifyMetadata()` only responsible for comparing response string with the expectation? 
Like `verifyMetadata(getMetadataResponse(0), StreamingNotnullToNull.metadataV0)`. 

By doing so we don't need to keep two versions of `verifyMetadata()`, and it's more readable.

Reply:
updated.